src/main/drivers/barometer/barometer_spl006.c

```diff
@@ -0,0 +1,240 @@
+
+    return pressure_cal + p_temp_comp;
+}
+
+bool spl006_calculate(baroDev_t * baro, int32_t * pressure, int32_t * temperature)
+{
+    UNUSED(baro);
+
+    if (pressure) *pressure = lrintf(spl006_compensate_pressure(spl006_pressure_raw, spl006_temperature_raw));
+    if (temperature) *temperature = lrintf(spl006_compensate_temperature(spl006_temperature_raw) * 100);
+
+    return true;
+}
+
+#define DETECTION_MAX_RETRY_COUNT   5
+static bool deviceDetect(busDevice_t * busDev)
+{
+    uint8_t chipId;
+    for (int retry = 0; retry < DETECTION_MAX_RETRY_COUNT; retry++) {
+        delay(100);
+        bool ack = busRead(busDev, SPL006_CHIP_ID_REG, &chipId);
+        if (ack && chipId == SPL006_DEFAULT_CHIP_ID) return true;
+    };
+
+    return false;
+}
+
+static bool read_calibration_coefficients(baroDev_t *baro) {
+    uint8_t sstatus;
+    if (!(busRead(baro->busDev, SPL006_MODE_AND_STATUS_REG, &sstatus) && (sstatus & SPL006_MEAS_CFG_COEFFS_RDY)))
+        return false;   // error reading status or coefficients not ready
+
+    uint8_t caldata[SPL006_CALIB_COEFFS_LEN];
+
+    if (!busReadBuf(baro->busDev, SPL006_CALIB_COEFFS_START, (uint8_t *)&caldata, SPL006_CALIB_COEFFS_LEN))
+        return false;
+
+    spl006_cal.c0 = (caldata[0] & 0x80 ? 0xF000 : 0) | ((uint16_t)caldata[0] << 4) | (((uint16_t)caldata[1] & 0xF0) >> 4);
+    spl006_cal.c1 = ((caldata[1] & 0x8 ? 0xF000 : 0) | ((uint16_t)caldata[1] & 0x0F) << 8) | (uint16_t)caldata[2];
+    spl006_cal.c00 = (caldata[3] & 0x80 ? 0xFFF00000 : 0) | ((uint32_t)caldata[3] << 12) | ((uint32_t)caldata[4] << 4) | (((uint32_t)caldata[5] & 0xF0) >> 4);
+    spl006_cal.c10 = (caldata[5] & 0x8 ? 0xFFF00000 : 0) | (((uint32_t)caldata[5] & 0x0F) << 16) | ((uint32_t)caldata[6] << 8) | (uint32_t)caldata[7];
+    spl006_cal.c01 = ((uint16_t)caldata[8] << 8) | ((uint16_t)caldata[9]);
+    spl006_cal.c11 = ((uint16_t)caldata[10] << 8) | (uint16_t)caldata[11];
+    spl006_cal.c20 = ((uint16_t)caldata[12] << 8) | (uint16_t)caldata[13];
+    spl006_cal.c21 = ((uint16_t)caldata[14] << 8) | (uint16_t)caldata[15];
+    spl006_cal.c30 = ((uint16_t)caldata[16] << 8) | (uint16_t)caldata[17];
+
+    return true;
+}
+
+static bool spl006_configure_measurements(baroDev_t *baro)
+{
+    uint8_t reg_value;
+
+    reg_value = SPL006_TEMP_USE_EXT_SENSOR | spl006_samples_to_cfg_reg_value(SPL006_TEMPERATURE_OVERSAMPLING);
+    if (!busWrite(baro->busDev, SPL006_TEMPERATURE_CFG_REG, reg_value)) goto nack_error;
+
+    reg_value = spl006_samples_to_cfg_reg_value(SPL006_PRESSURE_OVERSAMPLING);
+    if (!busWrite(baro->busDev, SPL006_PRESSURE_CFG_REG, reg_value)) goto nack_error;
+
+    reg_value = 0;
+    if (SPL006_TEMPERATURE_OVERSAMPLING > 8) reg_value |= SPL006_TEMPERATURE_RESULT_BIT_SHIFT;
+    if (SPL006_PRESSURE_OVERSAMPLING > 8) reg_value |= SPL006_PRESSURE_RESULT_BIT_SHIFT;
+    if (!busWrite(baro->busDev, SPL006_INT_AND_FIFO_CFG_REG, reg_value)) goto nack_error;
+
+    return true;
+
+nack_error:
```

Comment:
Generally `goto` should be avoided if possible. Here instead of `goto` you can just `return false`